http/src/main/java/io/micronaut/http/filter/FilterRunner.java

```diff
@@ -0,0 +1,966 @@
+                return true;
+            }
+            Object[] args = bindArgs(before);
+            if (before.passedOnContinuationArgIndex != -1) {
+                passedOnContinuation = (FilterContinuationImpl<?>) args[before.passedOnContinuationArgIndex];
+                if (completeOn != null) {
+                    passedOnContinuation.completeOn = completeOn;
+                }
+            }
+            Object returnValue = before.method.invoke(before.bean, args);
+
+            boolean proceed = before.returnHandler.handle(this, returnValue, passedOnContinuation);
+            if (passedOnContinuation != null && proceed) {
+                throw new AssertionError("handleFilterReturn should never return true if there is a continuation");
+            }
+            return proceed;
+        } catch (Throwable e) {
+            if (passedOnContinuation == null) {
+                failure = e;
+                responseNeedsProcessing = true;
+                workResponse();
+            } else {
+                passedOnContinuation.forwardResponse(null, e);
+            }
+            return false;
+        }
+    }
+
+    private <T> boolean invokeAfter(FilterMethod<T> after) {
+        try {
+            if (after.filterCondition != null && !after.filterCondition.test(this)) {
+                return true;
+            }
+            Object returnValue = after.method.invoke(after.bean, bindArgs(after));
+            return after.returnHandler.handle(this, returnValue, null);
+        } catch (Throwable e) {
+            failure = e;
+            responseNeedsProcessing = true;
+        }
+        return true;
+    }
+
+    private Object[] bindArgs(FilterMethod<?> filterMethod) {
+        Object[] args = new Object[filterMethod.argBinders.length];
+        for (int i = 0; i < args.length; i++) {
+            args[i] = filterMethod.argBinders[i].bind(this);
+        }
+        return args;
+    }
+
+    @Internal
+    public static <T> FilterMethod<T> prepareFilterMethod(T bean, ExecutableMethod<T, ?> method, boolean isResponseFilter, FilterOrder order) throws IllegalArgumentException {
+        return prepareFilterMethod(bean, method, method.getArguments(), method.getReturnType().asArgument(), isResponseFilter, order);
+    }
+
+    @Internal
+    public static <T> FilterMethod<T> prepareFilterMethod(T bean, ExecutableMethod<T, ?> method, Argument<?>[] arguments, Argument<?> returnType, boolean isResponseFilter, FilterOrder order) throws IllegalArgumentException {
+        FilterArgBinder[] fulfilled = new FilterArgBinder[arguments.length];
+        Predicate<FilterRunner> filterConditionAlwaysTrue = runner -> true;
+        Predicate<FilterRunner> filterCondition = filterConditionAlwaysTrue;
+        boolean skipOnError = isResponseFilter;
+
+        int passedOnContinuationArgIndex = -1;
+        for (int i = 0; i < arguments.length; i++) {
+            Argument<?> argument = arguments[i];
```

Comment:
Extract a method to fill the value to avoid duplicating `fulfilled[i] = `

Reply:
it can't be extracted because some parameter types also change the other local variables, for example having a Throwable parameter type changes the filterCondition